pkg/ingester/otlp/convert.go

```diff
@@ -168,59 +176,59 @@ func (p *profileBuilder) addfunc(s string) uint64 {
 	return idx
 }
 
-func serviceNameFromSample(p *otelProfile.Profile, sample *otelProfile.Sample) string {
+func serviceNameFromSample(sample *otelProfile.Sample, dictionary *otelProfile.ProfilesDictionary) string {
 	for _, attributeIndex := range sample.AttributeIndices {
-		attribute := p.AttributeTable[attributeIndex]
+		attribute := dictionary.AttributeTable[attributeIndex]
 		if attribute.Key == serviceNameKey {
 			return attribute.Value.GetStringValue()
 		}
 	}
 	return ""
 }
```

Comment:
I didn't look to deep, but it feels like a major change in the OTel spec, where we add a `Dictionary` member (to share parts across resource profiles). In the code we assume that the member is always present, and we hit nil deference the first time we access it – on this line.

Although it is an experimental API in dev, let's handle it more gracefully: if specs don't say anything about the fallback, we should drop the profile/request.